tabbycat/standings/teams.py

```diff
@@ -175,13 +175,19 @@ def get_annotation(self, round=None):
         annotation_filter = Q(
             debateteam__teamscore__ballot_submission__confirmed=True,
             debateteam__debate__round__stage=Round.STAGE_PRELIMINARY,
-            debateteam__teamscore__forfeit=False
+            debateteam__teamscore__forfeit=False,
+            debateteam__speakerscore__position__lte=self.tournament.last_substantive_position,
+            debateteam__speakerscore__ghost=False
         )
         if round is not None:
-            annotation_filter &= Q(debateteam__speakerscore__position__lte=round.tournament.last_substantive_position)
+            annotation_filter &= Q(debateteam__debate__round__seq__lte=round.seq)
 
         return Avg('debateteam__speakerscore__score', filter=annotation_filter)
 
+    def get_annotated_queryset(self, queryset, column_name, round=None):
+        self.tournament = queryset.first().tournament
```

Comment:
If the query set is empty, there is a bigger problem :)

Reply:
You're right. I added that in addition to the round check to have our bases covered. If the query set is empty (_i.e._ no teams), then there wouldn't be any reply speeches. Thus, with both failing, there won't be the reply filter. Hope that's better :)

Reply:
Makes sense! If I understood right: In this edge case, the filter is moot to an empty queryset anyway? Maybe a comment noting this would be helpful, since the rationale isn't obvious from the logic?

Reply:
Done. You understood correctly. Also I'm not good at commenting my code :)